pkg/trace/agent/agent_test.go

```diff
@@ -114,6 +114,46 @@ func TestProcess(t *testing.T) {
 		assert.Equal("SELECT name FROM people WHERE age = ? AND extra = ?", span.Meta["sql.query"])
 	})
 
+	t.Run("ReplacerMetrics", func(t *testing.T) {
+		cfg := config.New()
+		cfg.Endpoints[0].APIKey = "test"
+		cfg.ReplaceTags = []*config.ReplaceRule{
+			{
+				Name: "request.zipcode",
+				Re:   regexp.MustCompile(".*"),
+				Repl: "...",
+			},
+			{
+				Name: "*",
+				Re:   regexp.MustCompile("1337"),
+				Repl: "5555",
+			},
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		agnt := NewTestAgent(ctx, cfg, telemetry.NewNoopCollector())
+		defer cancel()
+
+		now := time.Now()
+		span := &pb.Span{
+			TraceID:  1,
+			SpanID:   1,
+			Resource: "resource",
+			Type:     "web",
+			Start:    now.Add(-time.Second).UnixNano(),
+			Duration: (500 * time.Millisecond).Nanoseconds(),
+			Metrics:  map[string]float64{"request.zipcode": 12345, "request.secret": 1337},
+			Meta:     map[string]string{},
```

Comment:
let's also put something in here and make sure it doesn't get obfuscated.

Also, what happens if there's a `request.secret` in the Meta map already...?

Reply:
It will get overwritten (although I'm not sure what the behavior in the UI is if you have duplicate keys with different values across `meta` and `metrics`)

Reply:
Just tried to test this and the go tracer actually doesn't allow duplicate tag keys, it's last one wins (so hypothetically I think it should be safe to assume that it's not there, I'll ask the other tracers though to be sure)